File: java/src/strategy/implementations/Dummy.java
package strategy.implementations;

import strategy.Answer;
import strategy.Code;
import strategy.IMastermindStrategy;
import strategy.T;

/**
 * A dummy strategy that always returns the same code.
 * 
 * E.g. CODE_LENGTH = 4 then the returned guess is always T0,T1,T2,T3.
 */
public class Dummy implements IMastermindStrategy {

	private final int CODE_LENGTH;
	
	public Dummy(int codeLength) {
		this.CODE_LENGTH = codeLength;
	}
	
	@Override
	public Code guess(Answer answer) {
		T[] t = new T[this.CODE_LENGTH];
		for(int i = 0; i < this.CODE_LENGTH; i++) {
			t[i] = T.values()[i];
		}
		return new Code(t);
	}

	@Override
	public Code reset() {
		T[] t = new T[this.CODE_LENGTH];
		for(int i = 0; i < this.CODE_LENGTH; i++) {
			t[i] = T.values()[i];
		}
		return new Code(t);
	}
	
	@Override
	public String toString() {
		return "Dummy";
	}

}

File: haskell/Strategies/Dummy.hs
-- | The dummy strategy always guesses the same code consisting of the buttons
-- T0,...,Tx where x is the length.

module Strategies.Dummy (dummy) where

import Strategies.Util

dummy = Strategy 
  { initialize   = initialize' 
  , extractGuess = extractGuess'
  , updateState  = updateState'
  }

-- | Memorize the code length.
initialize' :: Int -> Int
initialize' = id

-- | Always return the same guess.
extractGuess' :: Int -> Code
extractGuess' length = map (buttons !!) [0..length-1]

-- | Do not update any state.
updateState' :: Answer -> Int -> Int
updateState' = flip const 
